### ISSUE TITLE
Allow tools to handle InfiniteWavefunctionRight files

### ISSUE THREAD

**Issue by jesse-osborne** [mptoolkit/mptoolkit#19]
Currently, (probably) only `mp-info` can handle InfiniteWavefunctionRight files. Other tools should be able to handle them easily, since the underlying functions which the tool calls would either be defined for InfiniteWavefunctionRight already, or we could convert to left-canonical form and call the same function.

On the other hand, the only reason for saving right-canonical form wavefunctions is so they can be used for IBC and EA wavefunction, this is for two reasons:
- So we can ensure that the same A-matrices for the right-canonical form are used each time we call the program, which isn't necessarily guaranteed if we convert from left-canonical form.
- So we can save the left and right boundaries of an IBC/EA file separately, instead of having to incorporate them into each file.

Since the only way (at the moment) to get a InfiniteWavefunctionRight file is to convert an InfiniteWavefunctionLeft into one using `mp-right-canonicalize`, there may not really be any reason for wanting to, e.g., calculate an expectation value for the right-canonical form, since we will always have the left canonical form anyway.

**Comment by ianmccul** [comment]
Yeah, the easiest way to handle this is to automatically convert, when it makes sense to do so.

Does having the boundary wavefunctions in an IBC by reference to a file actually work?  This is something that I couldn't test as I was writing the code.
